admin/failures/overview/endpoint.js

```diff
@@ -51,6 +46,20 @@ module.exports = (purchases) => (req, res) => {
     });
 };
 
+function getLatestTimestamp(order){
+    if(!order.data.errors || !order.data.errors.length){
+        return order.data.viewModel.date;
+    }
+    return order.data.errors[order.data.errors.length - 1].at;

```

Comment:
Now that you have a `getLastElement` helper function, you can use it here to make the code even simpler:

```js
const lastError = getLastElement(order.data.errors);
if(lastError) {
    return lastError.at;
}
return order.data.viewModel.date;
```

This is *more lines* but conceptually simpler and easier to read because we separate the two levels of abstraction:

- mechanics of getting last item of a list that may or may not be defined
- presentational logic of getting a date or doing a fallback to a view model date